src/odemis/gui/cont/acquisition/cryo_acq.py

```diff
@@ -286,6 +289,29 @@ def _on_export_data_done(self, future):
         data = future.result()
         self._display_acquired_data(data)
 
+    def _add_feature_info_filename(self):
+        """
+        Add feature name, feature status and the counter at the end of the filename.
+        """
+        local_tab = self._tab_data.main.getTabByName("cryosecom-localization")
+        feature_name = local_tab.tab_data_model.main.currentFeature.value.name.value
+        feature_status = local_tab.tab_data_model.main.currentFeature.value.status.value
+        feature_pattern = f"{feature_name}-{feature_status}"
+
+        if 'Feature' in self._config.fn_ptn:
+            # If feature pattern is present in filename
+            # Replace Feature and everything after with the current feature pattern and the updated count
+            base_name = re.sub(r'(Feature).*', f'{feature_pattern}-0', self._config.fn_ptn)
+        else:
+            # If feature pattern not present in filename add it
+            # Add the feature pattern and count at the end of the given filename
+            base_name = re.sub('{cnt}', f'-{feature_pattern}-0', self._config.fn_ptn)
+
+        # update the filename for the current acquisition
+        list_names = os.listdir(self._config.pj_last_path)
+        name = make_unique_name(f'{base_name}{self._config.last_extension}', list_names)
+        self._filename.value = os.path.join(self._config.pj_last_path, name)
```

Comment:
I think that instead of basing yourself on the pattern, you should take the expected filename (as argument), compute the basename and extension (using util.split_ext) and then create a new name by creating a pattern which only has fixed elements and `{cnt}`, like: `f"{basename}-{feature_name}-{feature_status}-\{cnt\}"`. Then  util.filename.create_filename() will automatically find the right count based on the current files in the directory.

You can return the filename to the caller.